### DISCUSSION TITLE
How to use a 3d volume scene in mitsuba3

### DISCUSSION THREAD

**Discussion by qsh-zh** [mitsuba-renderer/mitsuba3#472] [Q&A]
I am new to mitsuba3. It seems the scene is stored in a customized XML format. If I have a sdf in 3d tensor, with each element representing the distance to the surface, what is the best practice to port the 3d tensor in mitsuba3? 

Thanks,

**Comment by njroussel** [comment]
Hi @qsh-zh 

There are two ways you can describe scenes in Mitusba. The first is the XML description as you mentioned, the second is a Python direction as explained here: https://mitsuba.readthedocs.io/en/stable/src/rendering/multi_view_rendering.html#Loading-a-scene.

I'm not quite sure what you're trying to achieve. Currently, Mitsuba doesn't have a SDF grid `Shape` (we are implementing one which should be available soon). You would need to define the shape yourself  (in C++/CUDA) or convert it using some third-party tool to a Mesh type supported by Mitsuba (OBJ/PLY).

If you just want the tensor in Mitsuba you can write `mi.TensorXf(sdf_tensor)`, where `sdf_tensor` is a Numpy `ndarray`.

If you're just getting started, I'd highly recommend you read through some of the tutorials and documentation on https://mitsuba.readthedocs.io/en/stable/index.html and https://drjit.readthedocs.io/en/latest/.
